### PR TITLE
Upgrade to Gradle 8

### PR DESCRIPTION
**Pull request by AndiCover**

### CODE REVIEW

**Comment by cherylking** [comment]
@AndiCover Hello Andi, thank you for getting this support for Gradle 8 work going. We have issue #799 open for this. If you are a first time contributer to the OpenLiberty GH organization, we will need you to submit the CLA. The instructions are [here](https://github.com/OpenLiberty/ci.gradle/blob/main/CONTRIBUTING.md).

**Comment by AndiCover** [comment]
Hi @cherylking
Thanks, I forgot to link the issue.
I submitted the signed CLA today.

**Comment by cherylking** [comment]
Looks like 3 tests are failing. After looking at the logs, two that I was able to identify are:

* arquillian-tests - is complaining about a problem with explicit/implicit dependencies between tasks
* loose-war-with-loose-jar - 
```
2023-04-26T15:38:16.8649386Z     Caused by: groovy.lang.MissingPropertyException: Could not get unknown property 'getArchiveBaseName' for task ':ejb-war:war' of type org.gradle.api.tasks.bundling.War.
```

I'm still looking for what the third test failure is. But overall, the majority of tests passed.

**Comment by AndiCover** [comment]
Hi @cherylking,
I can't get the arquillian tests to work on my machine. Can you provide the necessary steps? 
I also tried it with the previous version.

I am working with IntelliJ and OpenJDK 17.0.1.
Because the 1.8.24-SNAPSHOT of ci.commons is needed, I also cloned this repository and published the snapshot to maven local. Now I was able to build the project and run most of the tests.

The arquillian-tests fail with the following exception:
`

        org.arquillian.example.GreeterIT > classMethod FAILED
        org.jboss.arquillian.container.spi.client.container.LifecycleException: Could not start container
        at app//io.openliberty.arquillian.managed.WLPManagedContainer.start(WLPManagedContainer.java:278)
        at app//org.jboss.arquillian.container.impl.ContainerImpl.start(ContainerImpl.java:179)
        at app//org.jboss.arquillian.container.impl.client.container.ContainerLifecycleController$8.perform(ContainerLifecycleController.java:137)
        at app//org.jboss.arquillian.container.impl.client.container.ContainerLifecycleController$8.perform(ContainerLifecycleController.java:133)
        at app//org.jboss.arquillian.container.impl.client.container.ContainerLifecycleController.forContainer(ContainerLifecycleController.java:208)
        at app//org.jboss.arquillian.container.impl.client.container.ContainerLifecycleController.startContainer(ContainerLifecycleController.java:133)
        at java.base@11.0.17/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base@11.0.17/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base@11.0.17/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base@11.0.17/java.lang.reflect.Method.invoke(Method.java:566)
        at app//org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
        at app//org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
        at app//org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
        at app//org.jboss.arquillian.container.impl.client.ContainerDeploymentContextHandler.createContainerContext(ContainerDeploymentContextHandler.java:54)
        at java.base@11.0.17/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base@11.0.17/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base@11.0.17/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base@11.0.17/java.lang.reflect.Method.invoke(Method.java:566)
        at app//org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
        at app//org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
        at app//org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
        at app//org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:105)
        at app//org.jboss.arquillian.core.impl.EventImpl.fire(EventImpl.java:62)
        at app//org.jboss.arquillian.container.impl.client.container.ContainerLifecycleController$2.perform(ContainerLifecycleController.java:70)
        at app//org.jboss.arquillian.container.impl.client.container.ContainerLifecycleController$2.perform(ContainerLifecycleController.java:64)
        at app//org.jboss.arquillian.container.impl.client.container.ContainerLifecycleController.forEachSuiteContainer(ContainerLifecycleController.java:181)
        at app//org.jboss.arquillian.container.impl.client.container.ContainerLifecycleController.startSuiteContainers(ContainerLifecycleController.java:64)
        at java.base@11.0.17/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base@11.0.17/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base@11.0.17/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base@11.0.17/java.lang.reflect.Method.invoke(Method.java:566)
        at app//org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
        at app//org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
        at app//org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
        at app//org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
        at app//org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:105)
        at app//org.jboss.arquillian.core.impl.EventImpl.fire(EventImpl.java:62)
        at app//org.jboss.arquillian.container.test.impl.client.ContainerEventController.execute(ContainerEventController.java:83)
        at java.base@11.0.17/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base@11.0.17/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base@11.0.17/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base@11.0.17/java.lang.reflect.Method.invoke(Method.java:566)
        at app//org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
        at app//org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
        at app//org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
        at app//org.jboss.arquillian.test.impl.TestContextHandler.createSuiteContext(TestContextHandler.java:69)
        at java.base@11.0.17/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base@11.0.17/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base@11.0.17/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base@11.0.17/java.lang.reflect.Method.invoke(Method.java:566)
        at app//org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
        at app//org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
        at app//org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
        at app//org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:105)
        at app//org.jboss.arquillian.test.impl.EventTestRunnerAdaptor.beforeSuite(EventTestRunnerAdaptor.java:71)
        at app//org.jboss.arquillian.junit.AdaptorManager.initializeAdaptor(AdaptorManager.java:23)
        at app//org.jboss.arquillian.junit.AdaptorManagerWithNotifier.initializeAdaptor(AdaptorManagerWithNotifier.java:19)
        at app//org.jboss.arquillian.junit.Arquillian.run(Arquillian.java:109)
        at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.runTestClass(JUnitTestClassExecutor.java:110)
        at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:58)
        at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:38)
        at org.gradle.api.internal.tasks.testing.junit.AbstractJUnitTestClassProcessor.processTestClass(AbstractJUnitTestClassProcessor.java:62)
        at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:51)
        at java.base@11.0.17/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base@11.0.17/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base@11.0.17/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base@11.0.17/java.lang.reflect.Method.invoke(Method.java:566)
        at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
        at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
        at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
        at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
        at com.sun.proxy.$Proxy2.processTestClass(Unknown Source)
        at org.gradle.api.internal.tasks.testing.worker.TestWorker$2.run(TestWorker.java:176)
        at org.gradle.api.internal.tasks.testing.worker.TestWorker.executeAndMaintainThreadName(TestWorker.java:129)
        at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:100)
        at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:60)
        at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:56)
        at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:133)
        at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:71)
        at app//worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
        at app//worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)

        Caused by:
        org.jboss.arquillian.container.spi.client.container.LifecycleException: Unable to retrieve connector address for localConnector of started VM. vmid: null connectedVM: null
            at app//io.openliberty.arquillian.managed.WLPManagedContainer.start(WLPManagedContainer.java:273)
            ... 80 more
[AUDIT   ] CWWKE0055I: Server shutdown requested on Thursday, April 27, 2023 at 9:25 AM. The server LibertyProjectServer is shutting down.
`

For some reason the configureArquillian task does not run.

**Comment by cherylking** [comment]
@AndiCover I will try to run the `arquillian-tests` locally today and see what I find. Do you know why the other test is complaining about the `unknown property getArchiveBaseName`?

File: src/test/resources/loose-war-with-loose-jar/ejb-war/build.gradle [hunk 1]
```diff
@@ -25,7 +25,7 @@ dependencies {
 }
 
 war {
-    archiveName = baseName + '.' + extension
+    archiveFileName = getArchiveBaseName.get() + '.' + getArchiveExtension().get()
```
Comment (cherylking) [inline src/test/resources/loose-war-with-loose-jar/ejb-war/build.gradle:28]: I think this is missing the `()` after `getArchiveBaseName`.

Comment (AndiCover) [reply src/test/resources/loose-war-with-loose-jar/ejb-war/build.gradle]: Thanks, I missed that. Updated the PR.

**Comment by cherylking** [comment]
@AndiCover I seem to have fixed the arquillian test locally on my machine. Here is what I changed:

* In `src/test/resources/arquillian-tests/build.gradle`, remove the `dependsOn 'compileJava'` that you added as it didn't resolve the test case problem.
* In `src/test/resources/aquillian-tests/settings.gradle`, add the following lines (for some reason it was trying to build a jar file for each sub-module, but there is nothing to be built in the sub-modules):

```
startParameter.excludedTaskNames += ':verifyMainAppManaged:build'
startParameter.excludedTaskNames += ':skipWithXmlManaged:build'
startParameter.excludedTaskNames += ':skipWithoutXmlManaged:build'
startParameter.excludedTaskNames += ':skipWithXmlRemote:build'
startParameter.excludedTaskNames += ':skipWithoutXmlRemote:build'
```

**Comment by AndiCover** [comment]
@cherylking Thanks!
Unfortunatelly, I still get the same exception. Am I missing some prerequesites?
I can update the PR but I am still not able to run all tests locally. I will try to run them on another device in the next days.

**Comment by cherylking** [comment]
> @cherylking Thanks! Unfortunatelly, I still get the same exception. Am I missing some prerequesites? I can update the PR but I am still not able to run all tests locally. I will try to run them on another device in the next days.

I was able to run locally and get the same errors that happen on the build system. I don't know why you are hitting something different locally.

**Comment by cherylking** [comment]
Looks like we have some green builds! I just resubmitted the failing ones to see if some might turn green (we have timing issues, locking issues, etc that can happen). 

I think the changes overall are good though. I will look to approve and merge later today or tomorrow morning. Thanks again for your contribution @AndiCover !

File: src/main/groovy/io/openliberty/tools/gradle/tasks/AbstractServerTask.groovy [hunk 2]
```diff
@@ -653,9 +653,9 @@ abstract class AbstractServerTask extends AbstractLibertyTask {
                 }
                 application.appendNode('appsDirectory', appDir)
                 if (server.looseApplication) {
-                    application.appendNode('applicationFilename', appObj.archiveName + '.xml')
+                    application.appendNode('applicationFilename', appObj.getArchiveFileName().get() + '.xml')
                 } else {
-                    application.appendNode('applicationFilename', appObj.archiveName)
+                    application.appendNode('applicationFilename', appObj.archiveFileName)
```
Comment (cherylking) [inline src/main/groovy/io/openliberty/tools/gradle/tasks/AbstractServerTask.groovy:658]: I see quite a few places where you are using `archiveFileName` directly and other places where you use `getArchiveFileName().get()`. Is the difference intentional?

File: src/test/resources/liberty-package-test/liberty-package-looseApplication.gradle [hunk 3]
```diff
@@ -38,12 +38,12 @@ ext {
 }
 
 war {
-    archiveName = baseName + '.' + extension
+    archiveFileName = archiveBaseName.get() + '.' + getArchiveExtension().get()
```
Comment (cherylking) [inline src/test/resources/liberty-package-test/liberty-package-looseApplication.gradle:41]: Should this use `getArchiveBaseName()`?

**Review by cherylking — COMMENTED** [review]
@AndiCover See review comments.

**Comment by cherylking** [comment]
The changes to `src/test/resources/aquillian-tests/settings.gradle` seem to have been lost in the latest commits.

**Comment by AndiCover** [comment]
@cherylking Thanks!
Everything should be included now and I will run the tests once again locally.

**Review by cherylking — APPROVED** [review]

**Pull request merged**